Prepare for Java 9 modules: set automatic module name

Currently, one cannot easily use this library on a modularized Java project. See the error:
```
my-app/build/jlinkbase/tmpjars/java-multiaddr-v1.3.0/module-info.java:2: error: '{' expected
open module java-multiaddr-v1.3.0 {
                ^
1 error

> Task :createDelegatingModules FAILED
```

This can be easily fixed. We can add a single entry to `META-INF/MANIFEST.MF` that will allow this lib to be used by modularized Java projects.

See http://branchandbound.net/blog/java/2017/12/automatic-module-name/

--- a/pom.xml
+++ b/pom.xml
@@ -89,6 +89,9 @@
 						<manifest>
 							<addDefaultImplementationEntries>true</addDefaultImplementationEntries>
 						</manifest>
+						<manifestEntries>
+							<Automatic-Module-Name>io.ipfs.multiaddr</Automatic-Module-Name>
+						</manifestEntries>
 					</archive>
 				</configuration>
 			</plugin>

--- a/pom.xml
+++ b/pom.xml
@@ -4,8 +4,7 @@
 
     <groupId>com.github.multiformats</groupId>
     <artifactId>java-multiaddr</artifactId>
-	<version>v1.3.1</version>
-	<packaging>jar</packaging>
+	<version>v1.3.2</version>
 
 	<name>multiaddr</name>
 	<url>https://github.com/multiformats/java-multiaddr</url>

